File: backend/src/main/java/app/models/Book.java
package app.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class Book {

    @OneToMany(mappedBy = "book")
    Set<BookStep> bookSteps = new HashSet<>();

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String title;
    private float price;
    private int pages;
    private String isbn;
    private String nuart;
    private String note;
    private String summary;
    private String hook;
    private String marketing;
    @ManyToMany(cascade = {CascadeType.MERGE})
    @JoinTable(
            name = "book_author",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "author_id")
    )
    private Set<Author> authors = new HashSet<>();
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "status_id")
    private Status globalStatus;
    @Column(name = "cover_filename")
    private String coverFileName;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;
    @Column(name = "release_date")
    private LocalDate releaseDate;

    public void addAuthor(Author author) {
        authors.add(author);
        author.getBooks().add(this);
    }

    public void removeAuthor(Author author) {
        authors.remove(author);
        author.getBooks().remove(this);
    }

}

File: frontend/src/app/shared/models/Book.ts
import {Author} from './Author';
import {BookStep} from './BookStep';

export class Book {
  id!: number;
  title!: string;
  isbn!: string;
  nuart!: string;
  pages!: number;
  price!: number;
  summary: string | null = null;
  hook: string | null = null;
  marketing: string | null = null;
  note: string | null = null;
  coverUrl: string | null = null;
  authors: Author[] | null = null;
  releaseDate: Date | null = null;
  bookSteps: BookStep[] | null = null;

  constructor(data: Partial<Book> = {}) {
    Object.assign(this, data);
  }
}
